File: flutter_app/lib/core/domain/entities/user_full.dart
import 'package:flutter_group_chat_app_with_firebase/core/domain/entities/user_public.dart';

class UserFull extends UserPublic {
  final String? fcmToken;

  UserFull ({required String uid, required this.fcmToken, required String firstName, required String lastName})
      : super(uid: uid, firstName: firstName, lastName: lastName);
}

File: functions/src/domain/entities/user-full.ts
import {UserPublic} from "./user-public";


export class UserFull extends UserPublic{

    constructor(
        uid: string,
        firstName: string,
        lastName: string,
        agoraUid: number,
        public readonly fcmToken: string,
    ) {
        super(uid, firstName, lastName, agoraUid);
    }
}
